src/main/plugin/iso19139.ca.HNAP/schematron/schematron-rules-common.sch

```diff
@@ -403,6 +403,21 @@
 
     </sch:rule>
 
+    <sch:rule context="//gmd:distributionInfo/gmd:MD_Distribution/gmd:transferOptions/gmd:MD_DigitalTransferOptions/gmd:onLine/gmd:CI_OnlineResource">
+        <sch:let name="protocolMissing" value="not(string(gmd:protocol/gco:CharacterString))" />
+    		<sch:let name="isNotPlaceholderProtocol" value="(string(gmd:protocol/gco:CharacterString) != 'WWW:DOWNLOAD-1.0-http--download')"/>
```

Comment:
I think this check can be removed, no? Below are checked the the valid values.

Reply:
I totally refined the validation. I used @ianwallen suggested approach and got good result. Here is what it looks likes from my debug console.

![image](https://user-images.githubusercontent.com/74916635/163050978-a7826951-cf5c-4a55-9949-9bc29a19cb33.png)